src/patches.js

```diff
@@ -103,29 +128,55 @@ export function applyPatches(draft, patches) {
         } else {
             let base = draft
             for (let i = 0; i < path.length - 1; i++) {
-                base = base[path[i]]
+                if (isMap(base.base)) {
+                    base = base.get(path[i])
+                } else {
+                    base = base[path[i]]
+                }
                 if (!base || typeof base !== "object")
                     throw new Error("Cannot apply patch, path doesn't resolve: " + path.join("/")) // prettier-ignore
             }
             const key = path[path.length - 1]
+
+            function replace(key, value) {
+                if (isMap(base.base)) {
+                    base.set(key, value)
+                    return
+                }
+                base[key] = value
+            }
+            function add(key, value) {
+                if (isMap(base.base)) {
+                    base.set(key, value)
+                    return
+                }
+                if (Array.isArray(base)) {
```

Comment:
nit: Probably want `isArray` check before `isMap` since it's more likely to be true